diff -ru flexipatch/dwm/config.mk patches/dwm/config.mk
--- a/config.mk	2026-06-13 00:49:29.282020120 +0800
+++ b/config.mk	2026-06-13 01:31:20.246346828 +0800
@@ -48,6 +48,9 @@
 # Uncomment this for the swallow patch / SWALLOW_PATCH
 XCBLIBS = -lX11-xcb -lxcb -lxcb-res
 
+# Add for xcursor patch
+XCURSOR = -lXcursor
+
 # This is needed for the winicon and tagpreview patches / BAR_WINICON_PATCH / BAR_TAGPREVIEW_PATCH
 #IMLIB2LIBS = -lImlib2
 
@@ -60,7 +63,7 @@
 
 # includes and libs
 INCS = -I${X11INC} -I${FREETYPEINC} ${YAJLINC} ${PANGOINC} ${BDINC}
-LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS} ${FREETYPELIBS}  ${XRENDER} ${MPDCLIENT} ${XEXTLIB} ${XCBLIBS} ${KVMLIB} ${PANGOLIB} ${YAJLLIBS} ${IMLIB2LIBS} $(BDLIBS) $(XILIB)
+LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS} ${FREETYPELIBS}  ${XRENDER} ${MPDCLIENT} ${XEXTLIB} ${XCBLIBS} ${KVMLIB} ${PANGOLIB} ${YAJLLIBS} ${IMLIB2LIBS} ${XCURSOR} $(BDLIBS) $(XILIB)
 
 # flags
 CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700L -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
diff -ru flexipatch/dwm/drw.c patches/dwm/drw.c
--- a/drw.c	2026-06-13 00:49:29.926604845 +0800
+++ b/drw.c	2026-06-13 01:28:25.240846186 +0800
@@ -4,6 +4,7 @@
 #include <string.h>
 #include <X11/Xlib.h>
 #include <X11/Xft/Xft.h>
+#include <X11/Xcursor/Xcursor.h>
 
 #include "drw.h"
 #include "util.h"
@@ -469,14 +470,14 @@
 }
 
 Cur *
-drw_cur_create(Drw *drw, int shape)
+drw_cur_create(Drw *drw, const char *shape)
 {
 	Cur *cur;
 
 	if (!drw || !(cur = ecalloc(1, sizeof(Cur))))
 		return NULL;
 
-	cur->cursor = XCreateFontCursor(drw->dpy, shape);
+	cur->cursor = XcursorLibraryLoadCursor(drw->dpy, shape);
 
 	return cur;
 }
diff -ru flexipatch/dwm/drw.h patches/dwm/drw.h
--- a/drw.h	2026-06-13 00:49:29.926604845 +0800
+++ b/drw.h	2026-06-13 01:28:25.241206207 +0800
@@ -55,7 +55,7 @@
 );
 
 /* Cursor abstraction */
-Cur *drw_cur_create(Drw *drw, int shape);
+Cur *drw_cur_create(Drw *drw, const char *shape);
 void drw_cur_free(Drw *drw, Cur *cursor);
 
 /* Drawing context manipulation */
diff -ru flexipatch/dwm/dwm.c patches/dwm/dwm.c
--- a/dwm.c	2026-06-13 04:38:29.855813943 +0800
+++ b/dwm.c	2026-06-13 01:34:31.122842646 +0800
@@ -30,7 +30,6 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <X11/cursorfont.h>
 #include <X11/keysym.h>
 #include <X11/Xatom.h>
 #include <X11/Xlib.h>
@@ -40,6 +39,7 @@
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
 #include <X11/Xft/Xft.h>
+#include <X11/Xcursor/Xcursor.h>
 
 #include "drw.h"
 #include "util.h"
@@ -1055,7 +1055,7 @@
 drawbar(Monitor *m)
 {
 	Bar *bar;
-	
+
 	if (m->showbar)
 		for (bar = m->bar; bar; bar = bar->next)
 			drawbarwin(bar);
@@ -2289,14 +2289,16 @@
 	netatom[NetClientList] = XInternAtom(dpy, "_NET_CLIENT_LIST", False);
 	motifatom = XInternAtom(dpy, "_MOTIF_WM_HINTS", False);
 	/* init cursors */
-	cursor[CurNormal] = drw_cur_create(drw, XC_left_ptr);
-	cursor[CurResize] = drw_cur_create(drw, XC_sizing);
-	cursor[CurResizeBR] = drw_cur_create(drw, XC_bottom_right_corner);
-	cursor[CurResizeBL] = drw_cur_create(drw, XC_bottom_left_corner);
-	cursor[CurResizeTR] = drw_cur_create(drw, XC_top_right_corner);
-	cursor[CurResizeTL] = drw_cur_create(drw, XC_top_left_corner);
-	cursor[CurDragFact] = drw_cur_create(drw, XC_rightbutton);
-	cursor[CurMove] = drw_cur_create(drw, XC_fleur);
+	cursor[CurNormal] = drw_cur_create(drw, "left_ptr");
+	cursor[CurResize] = drw_cur_create(drw, "se-resize");
+	cursor[CurResizeBR] = drw_cur_create(drw, "bottom_right_corner");
+	cursor[CurResizeBL] = drw_cur_create(drw, "bottom_left_corner");
+	cursor[CurResizeTR] = drw_cur_create(drw, "top_right_corner");
+	cursor[CurResizeTL] = drw_cur_create(drw, "top_left_corner");
+	cursor[CurResizeHorzArrow] = drw_cur_create(drw, "sb_h_double_arrow");
+	cursor[CurResizeVertArrow] = drw_cur_create(drw, "sb_v_double_arrow");
+	cursor[CurIronCross] = drw_cur_create(drw, "cross");
+	cursor[CurMove]   = drw_cur_create(drw, "fleur");
 	/* init appearance */
 	scheme = ecalloc(LENGTH(colors), sizeof(Clr *));
 	for (i = 0; i < LENGTH(colors); i++)
